examples/tests/sdk_tests/expect/assert.main.w

```diff
@@ -0,0 +1,210 @@
+}
+
+// String tests
+
+let c: str? = nil;
+let d: str? = "hello";
+
+expect.equal("hello", "hello");
+expect.equal(d, "hello");
+expect.nil(c);
+expect.notNil(d);
+expect.notEqual(d, "world");
+expect.notEqual(d, 1);
+expect.notEqual(d, true);
+
+test "equal str" {
+    expect.equal("hello", "hello");
+    expect.equal(d, "hello");
+    expect.nil(c);
+    expect.notNil(d);
+    expect.notEqual(d, "world");
+    expect.notEqual(d, 1);
+    expect.notEqual(d, true);
+}
+
+// Bool tests
+
+let e: bool? = nil;
+let f: bool? = true;
+
+expect.equal(true, true);
+expect.equal(f, true);
+expect.nil(e);
+expect.notNil(f);
+expect.notEqual(f, false);
+
+test "equal bool" {
+    expect.equal(true, true);
+    expect.equal(f, true);
+    expect.nil(e);
+    expect.notNil(f);
+    expect.notEqual(f, false);
+}
+
+// List of str tests
+let g = ["hello"];
+
+expect.equal(g, ["hello"]);
+expect.notEqual(g, ["world"]);
+
+test "equal array of strings" {
+    expect.equal(g, ["hello"]);
+    expect.notEqual(g, ["world"]);
+}
+
+// List of numbers tests
+let h = [1];
+
+expect.equal(h, [1]);
+expect.notEqual(h, ["world"]);
+
+test "equal array of numbers" {
+    expect.equal(h, [1]);
+    expect.notEqual(h, ["world"]);
+}
+
+// Json object tests
+let obj = Json { key1: 1, key2: 2};
+
+expect.equal(obj, Json { key1: 1, key2: 2});
+expect.notEqual(obj, Json { key1: 1, key3: 3});
+
+test "equal objects" {
+    expect.equal(obj, Json { key1: 1, key2: 2});
+    expect.notEqual(obj, Json { key1: 1, key3: 3});
+}
+
+// Map tests
+
+let maps = { "hello" => 123, "world" => 99 };
+expect.equal(maps, { "hello" => 123, "world" => 99 });
+expect.notEqual(maps, { "hello" => 123, "world" => 100 });
+
+test "equal maps" {
+    expect.equal(maps, { "hello" => 123, "world" => 99 });
+    expect.notEqual(maps, { "hello" => 123, "world" => 100 });
+}
+
+
+// Set tests
+
+let mySet = Set<num> {1, 2, 3};
+
+expect.equal(mySet, Set<num> {1, 2, 3});
+expect.notEqual(mySet, Set<num> {1, 2, 3, 4});
+
+test "equal sets" {
+    expect.equal(mySet, Set<num> {1, 2, 3});
+    expect.notEqual(mySet, Set<num> {1, 2, 3, 4});
+}
+
+// DateTime tests
+// FIXME: Doesn't work
+// ERROR: unable to serialize immutable data object of type Datetime
+
+// examples/tests/sdk_tests/expect/target/test/assert.main.wsim.256800.tmp/.wing/preflight.js:319
+
+// let dt = datetime.fromIso("2023-07-18T20:18:25.177+03:00");
+
+// expect.equal(dt, datetime.fromIso("2023-07-18T20:18:25.177+03:00"));
+// expect.notEqual(dt, datetime.fromIso("2024-07-18T20:18:25.177+03:00"));
+
+// test "equal time" {
+//     expect.equal(dt, datetime.fromIso("2023-07-18T20:18:25.177+03:00"));
+//     expect.notEqual(dt, datetime.fromIso("2024-07-18T20:18:25.177+03:00"));
+// }
+
+// Range tests
+// Doesn't work, while expect.equal([1,2,3,4], [1,2,3,4]); works and the JS equivalent works too
+// 
+//ERROR: Values have same structure but are not reference-equal:
+
+// [
+//     1,
+//     2,
+//     3,
+//     4
+//   ]
```

Comment:
Don't really know why these tests are failing. Ideas?

Reply:
There is a [bug](https://github.com/winglang/wing/issues/4118) currently that fails assertions for any array returned by something. I'm not sure this happens all the time but it has happened whenever I tried something, and now it looks like yet another example.

Reply:
Thanks for the pointer!

I'm generally ok to merge this as is, mainly wanted to flag the cases which I've found to be not comparable right now:

1. DateTime
2. Range
3. Custom classes (in particular inflight classes)